internal/gh/pullrequest.go

```diff
@@ -211,3 +211,79 @@ func (c *Client) RepoPullRequests(ctx context.Context, opts RepoPullRequestOpts)
 		PullRequests: query.Repository.PullRequests.Nodes,
 	}, nil
 }
+
+type PullRequestByNumberInput struct {
+	Owner  string
+	Repo   string
+	Number int64
+}
+
+type Commit struct {
+	Oid string
+}
+
+func (c *Client) PullRequestMergeCommit(ctx context.Context, opts PullRequestByNumberInput) (*string, error) {
+	var query struct {
+		Repository struct {
+			PullRequest struct {
+				MergeCommit *Commit
+			} `graphql:"pullRequest(number: $number)"`
+		} `graphql:"repository(owner: $owner, name: $repo)"`
+	}
+
+	vars := map[string]any{
+		"owner":  githubv4.String(opts.Owner),
+		"repo":   githubv4.String(opts.Repo),
+		"number": githubv4.Int(opts.Number),
+	}
+	if err := c.query(ctx, &query, vars); err != nil {
+		return nil, errors.Wrap(err, "failed to query merge commit")
+	}
+	if query.Repository.PullRequest.MergeCommit == nil {
+		return nil, nil
+	}
+	return &query.Repository.PullRequest.MergeCommit.Oid, nil
+}
+
+type TimelineItemCloserCommit struct {
+	Commit struct {
+		Oid string
+	} `graphql:"... on Commit"`
+}
+
+type TimelineItemClosedEvent struct {
+	ClosedEvent struct {
+		Closer *TimelineItemCloserCommit
+	} `graphql:"... on ClosedEvent"`
+}
+
+func (c *Client) PullRequestFastForwardCommit(ctx context.Context, opts PullRequestByNumberInput) (*string, error) {
```

Comment:
@travigd, testing this locally in the GraphQL explorer is not showing the query working like when you showed it to me. I created a test repo and a PR that was merged with fast forward mode but it seems like for me the `closer` on the closing event is always `null` - I have looked at some alternatives, but we will want to discuss this.

Reply:
Yeah, let's chat. Will also say that we should probably try to do everything in a single API request rather than having two different ones.